docker-run-sync-rapid-pro-to-engagement-db.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+set -e
+
+PROJECT_NAME="$(<configurations/docker_image_project_name.txt)"
+IMAGE_NAME=$PROJECT_NAME-rapid-pro-to-engagement-db
+
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        --incremental-cache-volume)
+            INCREMENTAL_ARG=" --incremental-cache-volume /cache"
+            INCREMENTAL_CACHE_VOLUME_NAME="$2"
+            shift 2;;
+        --local-archive)
+            LOCAL_ARCHIVE_PATHS+=("$2")
+            shift 2;;
+        --)
+            shift
+            break;;
+        *)
+            break;;
+    esac
+done
+
+# Check that the correct number of arguments were provided.
+if [[ $# -ne 3 ]]; then
+    echo "Usage: $0 
+    [--incremental-cache-volume <incremental-cache-volume>] 
+    [--local-archive <local_archive>] : set a single option with argument, repeat it multiple times
+    <user> <google-cloud-credentials-file-path> <configuration-module>"
+    exit
+fi
+
+# Assign the program arguments to bash variables.
+USER=$1
+GOOGLE_CLOUD_CREDENTIALS_PATH=$2
+CONFIGURATION_MODULE=$3
+
+# Build an image for this pipeline stage.
+docker build -t "$IMAGE_NAME" .
+
+LOCAL_ARCHIVE_ARGS=""
+for LOCAL_ARCHIVE_PATH in "${LOCAL_ARCHIVE_PATHS[@]}"; do
+    IFS="=" # Setting equal sign as delimiter 
+    read -a strarr <<<"$LOCAL_ARCHIVE_PATH" # Reading str as an array of tokens separated by IFS 
+    local_archive_dir=$(basename ${strarr[1]})
+    LOCAL_ARCHIVE_ARGS+=" --local-archive /$local_archive_dir"
+done
+
+# Create a container from the image that was just built.
+CMD="pipenv run python -u sync_rapid_pro_to_engagement_db.py ${INCREMENTAL_ARG} ${LOCAL_ARCHIVE_ARGS} \
+    ${USER} /credentials/google-cloud-credentials.json ${CONFIGURATION_MODULE}"
+
+if [[ "$INCREMENTAL_ARG" ]]; then
+    container="$(docker container create -w /app --mount source="$INCREMENTAL_CACHE_VOLUME_NAME",target=/cache "$IMAGE_NAME" /bin/bash -c "$CMD")"
+else
+    container="$(docker container create -w /app "$IMAGE_NAME" /bin/bash -c "$CMD")"
+fi
+
+echo "Created container $container"
+container_short_id=${container:0:7}
+
+# Copy input data into the container
+echo "Copying $GOOGLE_CLOUD_CREDENTIALS_PATH -> $container_short_id:/credentials/google-cloud-credentials.json"
+docker cp "$GOOGLE_CLOUD_CREDENTIALS_PATH" "$container:/credentials/google-cloud-credentials.json"
+
+for LOCAL_ARCHIVE_PATH in "${LOCAL_ARCHIVE_PATHS[@]}"; do
```

Comment:
Sorry you had to go through this. Well done finding a solution.

Reply:
Thanks.